openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -369,12 +358,11 @@ def add_user_to_cohort(cohort, username_or_email):
             "user_id": user.id,
             "cohort_id": cohort.id,
             "cohort_name": cohort.name,
-            "previous_cohort_id": previous_cohort_id,
-            "previous_cohort_name": previous_cohort_name,
+            "previous_cohort_id": membership.previous_cohort_id,
+            "previous_cohort_name": membership.previous_cohort_name,
         }
     )
```

Comment:
@explorerleslie, is this a decision you can help with? The behavior of this event has changed slightly; it is no longer possible to emit `user_add_requested` with previous cohort information before the membership is saved. We'll have to either remove the previous cohort information, or emit the event after the membership is saved.

The previous lines containing a TODO have been removed, here's the conversation that had happened to that point:
```
TODO: with the new setup, we cannot emit a "requested" event containing
information about the previous cohort before calling save().

Decide how to best handle this
    -remove previous info from "requested" event
    -wait until after save so previous information is available
    -add another event type and emit both before and after
```

cahrens   Collaborator
cahrens added a note a day ago
Am I correct that the only loss in functionality is that we won't emit a user_add_requested event if the save operation failed? What are the conditions under which it would fail?

 efischer19   Collaborator
efischer19 added a note a day ago
My main concern arose after I wrote these inline comments and found this code (https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/course_groups/cohorts.py#L38), which seems to be emitting an "action happened" event, in contrast to "the "action requested" event here. Could this timing information be getting gathered and utilized elsewhere? If that's the case, our best course of action would be to move the previous_cohort information to that event, and continue to emit the "requested" event prior to attempting the save.

Who would be the best person to ask about this design? explorerleslie? someone from Analytics?